ui/app/AppLayouts/Onboarding/OnboardingLayout.qml

```diff
@@ -190,22 +190,33 @@ QtObject {
         KeysMainView {
             onButtonClicked: {
                 if (state === "importseed") {
-                    importSeedState.seedInputState = "newUser";
+                    importSeedState.seedInputState = "existingUser";
                     Global.applicationWindow.navigateTo("ImportSeed");
                 } else {
+                    importSeedState.seedInputState = "newUser";
                     Global.applicationWindow.navigateTo("GenKey");
                 }
             }
             onKeycardLinkClicked: {
                 Global.applicationWindow.navigateTo("KeycardFlowSelection");
             }
             onSeedLinkClicked: {
-                importSeedState.seedInputState = "existingUser";
-                Global.applicationWindow.navigateTo("ImportSeed");
+                if (state === "getkeys") {
+                    importSeedState.seedInputState = "newUser";
+                    state = "importseed";
+                } else {
+                    importSeedState.seedInputState = "existingUser";
+                    Global.applicationWindow.navigateTo("ImportSeed");
+                }
             }
             onBackClicked: {
-                if ((root.keysMainSetState === "connectkeys" && LoginStore.currentAccount.username !== "") || root.prevState === "LogIn") {
+                if (state === "importseed") {
+                    state = "getkeys";
+                } else if (((root.keysMainSetState === "connectkeys") && (LoginStore.currentAccount.username !== ""))
+                          || (root.prevState === "LogIn") || (state === "getkeys")) {
                     Global.applicationWindow.navigateTo("LogIn");
+                } else if (state === "importseed") {
+                    Global.applicationWindow.navigateTo("GenKey");
```

Comment:
Hm.. this will never happen no? In order to end up in this path the first condition must have failed, and that condition is the same as this one... Am I missing something here?

Reply:
you're right @PascalPrecht, removed!